### ISSUE TITLE
Error out CircleCI builds with merge conflicts

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/conda-smithy#316]
Ideally we would want to check for merge conflicts and error out CircleCI builds if present (instead of building and wasting CI time/confusing users). This behavior is something that other CIs like TravisCI and AppVeyor have builtin.

However, we can't really do that well without knowing what the base branch is. As it may not be `master` we can't use this [trick]( https://github.com/conda-forge/staged-recipes/pull/1486/files#diff-dd26a71e242b54ab0e4ecd211a910652R27 ). Instead we really need CircleCI to provide that [information]( https://discuss.circleci.com/t/add-environment-varible-circle-pr-base-branch/6189 ).

Once CircleCI does provide that information, we can update the code to do something like [this]( https://discuss.circleci.com/t/show-test-results-for-prospective-merge-of-a-github-pr/1662/8?u=jakirkham ). This issue is here to serve as a reminder to catch and error on merge conflicts once CircleCI makes it possible.

**Comment by pelson** [comment]
Are we able to get it from the refs that github exposes? https://gist.github.com/piscisaureus/3342247#gistcomment-608797

**Comment by jakirkham** [comment]
If you have time for a rabbit hole, please read this ( https://github.com/isaacs/github/issues/757 ). Maybe good for ✈️. 😉

**TL;DR:** Sadly no. Ideally GitHub would just delete the merge ref when there is a conflict so it would be an error to fetch the merge ref. However, they don't so we need to either check the GitHub API or know the base branch.

**Comment by jakirkham** [comment]
Alternatively we could solve this if we knew the branch in advance. In other words, sharing a use case with other cases needing to know the branch ( https://github.com/conda-forge/conda-smithy/issues/318 ).

**Comment by h-vetinari** [comment]
CircleCI is dead; closing as obsolete. Xref #2627
